boards/rp-pico/examples/pico_spi_sd_card.rs

```diff
@@ -273,7 +273,7 @@ fn main() -> ! {
     blink_signals(&mut led_pin, &mut delay, &BLINK_OK_LONG);
 
     info!("Init SD card controller...");
-    let mut cont = Controller::new(block, DummyTimesource::default());
+    let mut cont: Controller<_, _, 4, 4> = Controller::new(block, DummyTimesource::default());
```

Comment:
I don't really know why it is required since the consts have default value.

Surprisingly (to me) this is enough to make it compile without any complaint:

```suggestion
    let mut cont: Controller<_, _> = Controller::new(block, DummyTimesource::default());
```

Reply:
I think this nicely explains why this happens: https://gankra.github.io/blah/defaults-affect-inference/#default-type-parameters

As I understand it: When declaring the type of `cont` with `let mut cont: Controller<_, _>`, rustc will use the default values.

But when calling `Controller::new( ... )`, the default type parameters are not used to constrain the possible values of the type parameters of `Controller`. All 4 of them need to be determined through type inference, and there is nothing in the existing code which restricts the two numeric parameters.

Reply:
Thanks! Today I learn this!